packages/ai/core/util/detect-mimetype.ts

```diff
@@ -83,13 +85,58 @@ export const audioMimeTypeSignatures = [
   },
 ] as const;
 
+const getID3v2TagSize = (header: Uint8Array) => {
+  const size =
+    ((header[6] & 0x7f) << 21) |
+    ((header[7] & 0x7f) << 14) |
+    ((header[8] & 0x7f) << 7) |
+    (header[9] & 0x7f);
+
+  return size + 10; // add header size
+};
+
+const stripID3 = (arrayBuffer: Uint8Array) => {
+  const bytes = new Uint8Array(arrayBuffer);
+
+  if (
+    bytes[0] === 0x49 && // 'I'
+    bytes[1] === 0x44 && // 'D'
+    bytes[2] === 0x33 // '3'
+  ) {
+    const id3Size = getID3v2TagSize(bytes);
+    return bytes.slice(id3Size); // The raw MP3 starts here
+  }
+
+  return bytes; // No ID3 tag, return as-is
+};
+
 export function detectMimeType({
   data,
   signatures,
 }: {
   data: Uint8Array | string;
   signatures: typeof audioMimeTypeSignatures | typeof imageMimeTypeSignatures;
 }): (typeof signatures)[number]['mimeType'] | undefined {
+  // Handle MP3 files with ID3 tags
```

Comment:
What is this for? Can we do this more elegantly?

Reply:
So this was a bit of a rabbit hole! Turns out some mp3 files contain id3 tags at the start of the buffer (metadata basically) so i have to check for and strip that, otherwise the byte prefixes are different.

Reply:
ooh cool i love hacking with file encodings 👀

Reply:
interesting - looks like this is not going to work for base64

Reply:
btw, is this a separate bugfix that should be extracted into a standalone PR?